app/src/main/java/com/scouts/kitchenplaner/ui/viewmodel/CreateRecipeViewModel.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * KitchenPlanerApp is the android app frontend for the KitchenPlaner, a tool
+ * to cooperatively plan a meal plan for a campout.
+ * Copyright (C) 2023-2024 Arne Kuchenbecker, Antonia Heiming
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+package com.scouts.kitchenplaner.ui.viewmodel
+
+import android.net.Uri
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.scouts.kitchenplaner.model.entities.Ingredient
+import com.scouts.kitchenplaner.model.entities.IngredientGroup
+import com.scouts.kitchenplaner.model.entities.Recipe
+import com.scouts.kitchenplaner.model.usecases.CreateRecipe
+import com.scouts.kitchenplaner.ui.state.RecipeAllergenState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class CreateRecipeViewModel @Inject constructor(
+    private val createRecipe: CreateRecipe
+) : ViewModel() {
+    var recipeName by mutableStateOf("")
+    var uri by mutableStateOf<Uri?>(null)
+    var description by mutableStateOf("")
+    var calculatedFor by mutableStateOf(1)
```

Comment:
Why 1?

Reply:
We need a default value here and the other two obvious choices would be 0 or a negative number, however both of these kind of break the assumption that when calculating the amount needed to cook a recipe (later on) we can just do `<amount of ingredients> * <number of people> / <calculatedFor>`

Reply:
okay